test/test_rest_vol.c

```diff
@@ -8711,23 +8720,159 @@ test_write_dataset_data_verification(void)
+        TEST_ERROR
+    if (H5Sselect_hyperslab(mspace_id, H5S_SELECT_SET, start, stride, count, block) < 0)
+        TEST_ERROR
+
     start[0] = start[1] = start[2] = 0;
     stride[0] = stride[1] = stride[2] = 1;
     count[0]                          = 2;
     count[1]                          = dims[1];
     count[2]                          = 1;
     block[0] = block[1] = block[2] = 1;
+    if (H5Sselect_hyperslab(fspace_id, H5S_SELECT_SET, start, stride, count, block) < 0)
+        TEST_ERROR
+
+    if (H5Dwrite(dset_id, DATASET_DATA_VERIFY_WRITE_TEST_DSET_DTYPE, mspace_id, fspace_id, H5P_DEFAULT,
+                 write_buf) < 0) {
+        H5_FAILED();
+        printf("    couldn't write to dataset\n");
+        goto error;
+    }
+
+    if (H5Sclose(mspace_id) < 0)
+        TEST_ERROR
+    if (H5Sclose(fspace_id) < 0)
+        TEST_ERROR
+    if (H5Dclose(dset_id) < 0)
+        TEST_ERROR
+
+    if ((dset_id = H5Dopen2(file_id, "/" DATASET_TEST_GROUP_NAME "/" DATASET_DATA_VERIFY_WRITE_TEST_DSET_NAME,
+                            H5P_DEFAULT)) < 0) {
+        H5_FAILED();
+        printf("    couldn't open dataset\n");
+        goto error;
+    }
 
+    if ((fspace_id = H5Dget_space(dset_id)) < 0) {
+        H5_FAILED();
+        printf("    couldn't get dataset dataspace\n");
+        goto error;
+    }
+
+    if ((space_npoints = H5Sget_simple_extent_npoints(fspace_id)) < 0) {
+        H5_FAILED();
+        printf("    couldn't get dataspace num points\n");
+        goto error;
+    }
+
+    if (NULL == (read_buf = malloc((hsize_t)space_npoints * DATASET_DATA_VERIFY_WRITE_TEST_DSET_DTYPESIZE)))
+        TEST_ERROR
+
+#ifdef RV_CONNECTOR_DEBUG
+    puts("Verifying that the data that comes back is correct after writing to the dataset using a hyperslab "
+         "selection\n");
+#endif
+
+    if (H5Dread(dset_id, DATASET_DATA_VERIFY_WRITE_TEST_DSET_DTYPE, H5S_ALL, H5S_ALL, H5P_DEFAULT, read_buf) <
+        0) {
+        H5_FAILED();
+        printf("    couldn't read from dataset\n");
+        goto error;
+    }
+
+    if (memcmp(data, read_buf, data_size)) {
+        H5_FAILED();
+        printf("    hyperslab selection data (contiguous) verification failed\n");
+        goto error;
+    }
+
+    if (data) {
+        free(data);
+        data = NULL;
+    }
+
+    if (write_buf) {
+        free(write_buf);
+        write_buf = NULL;
+    }
+
+    if (read_buf) {
+        free(read_buf);
+        read_buf = NULL;
+    }
+
+#ifdef RV_CONNECTOR_DEBUG
+    puts("Writing to dataset using hyperslab selection - contiguous - non-zero offset\n");
+#endif
+
+    data_size = dims[1] * 2 * DATASET_DATA_VERIFY_WRITE_TEST_DSET_DTYPESIZE;
+
+    if (NULL == (write_buf = malloc(data_size)))
+        TEST_ERROR
+
+    for (i = 0; i < dims[1]; i++)
+        ((int *)write_buf)[i] = 68;
+    for (i = dims[1]; i < data_size / DATASET_DATA_VERIFY_WRITE_TEST_DSET_DTYPESIZE; i++)
+        ((int *)write_buf)[i] = 67;
```

Comment:
If I understand correctly, shouldn't the condition in the second for loop here also be `i < dims[1]` ? Otherwise it looks like the first loop sets half the elements in `write_buf` and then the second loop starts at an offset of `dims[1]` in `write_buf` and set the full number of elements, which would write `dims[1]` elements past the end of `write_buf`.

Reply:
Ah, my mistake, ignore. I see what you're doing here now.

Reply:
I am happy to ignore this comment.